### Neste script vamos analisar os dados de malha fundiária mais importantes para a Hidreletrica de Teles Pires

Vamos analisar os grandes números para Teles Pires  
Primeiro vamos ler os dados

In [6]:
#Importando pacotes
import pandas as pd
import numpy as np
import glob

In [7]:
#Definindo nome de arquivos e diretórios:
dir_tables = 'C:/Users/User/Dropbox (CPI)/EduardoMinsky/Analise_TellesPires/Analise_Current/FeaturesTable/'
my_list_files = glob.glob(dir_tables+'*.csv')

malha = ['assentamento','CAR','FloraNaoDest','SIGEF','ProtInt','UsoSus','TI']
my_list_files

['C:/Users/User/Dropbox (CPI)/EduardoMinsky/Analise_TellesPires/Analise_Current/FeaturesTable\\Assentamentos_SumOfValues_featureTable.csv',
 'C:/Users/User/Dropbox (CPI)/EduardoMinsky/Analise_TellesPires/Analise_Current/FeaturesTable\\CAR_SumOfValues_featureTable.csv',
 'C:/Users/User/Dropbox (CPI)/EduardoMinsky/Analise_TellesPires/Analise_Current/FeaturesTable\\FloraNaoDest_SumOfValues_featureTable.csv',
 'C:/Users/User/Dropbox (CPI)/EduardoMinsky/Analise_TellesPires/Analise_Current/FeaturesTable\\Sigef_SumOfValues_featureTable.csv',
 'C:/Users/User/Dropbox (CPI)/EduardoMinsky/Analise_TellesPires/Analise_Current/FeaturesTable\\SNUCProtInt_SumOfValues_featureTable.csv',
 'C:/Users/User/Dropbox (CPI)/EduardoMinsky/Analise_TellesPires/Analise_Current/FeaturesTable\\SNUCUsoSUS_SumOfValues_featureTable.csv',
 'C:/Users/User/Dropbox (CPI)/EduardoMinsky/Analise_TellesPires/Analise_Current/FeaturesTable\\TIHomolog_SumOfValues_featureTable.csv']

In [8]:
#Lendo os arquivos, alimentando  a lista de DFs e criando uma coluna com que tipo de malha é

pd_list = []
for i in range(len(my_list_files)):
    a = pd.read_csv(my_list_files[i],sep = ';')
    a['Malha'] = malha[i]
    pd_list.append(a)




## Agora vamos calcular os grandes números para cada malha fundiária!  
Nós sabemos que a analise de Teles Pires possui 66.846.626,00 milhões de dolares.

In [9]:
#Vamos passar a coluna de área para number. Estava string
pd_list[0]['AssentSum'] = pd_list[0]['AssentSum'].str.replace(',','.').astype('float')
pd_list[1]['CAR_Sum'] = pd_list[1]['CAR_Sum'].str.replace(',','.').astype('float')
pd_list[2]['FloraNDest'] = pd_list[2]['FloraNDest'].str.replace(',','.').astype('float')
pd_list[3]['SigefSum'] = pd_list[3]['SigefSum'].str.replace(',','.').astype('float')
pd_list[4]['ProtIntSum'] = pd_list[4]['ProtIntSum'].str.replace(',','.').astype('float')
pd_list[5]['Sus_Sum'] = pd_list[5]['Sus_Sum'].str.replace(',','.').astype('float')
pd_list[6]['TI_Sum'] = pd_list[6]['TI_Sum'].str.replace(',','.').astype('float')


#Vamos passar a coluna de geometria para number. Estava string
pd_list[0]['area_geom'] = pd_list[0]['area_geom'].str.replace(',','.').astype('float')
pd_list[1]['area_geom'] = pd_list[1]['area_geom'].str.replace(',','.').astype('float')
pd_list[2]['area_geom'] = pd_list[2]['area_geom'].str.replace(',','.').astype('float')
pd_list[3]['area_geom'] = pd_list[3]['area_geom'].str.replace(',','.').astype('float')
pd_list[4]['area_geom'] = pd_list[4]['area_geom'].str.replace(',','.').astype('float')
pd_list[5]['area_geom'] = pd_list[5]['area_geom'].str.replace(',','.').astype('float')
pd_list[6]['area_geom'] = pd_list[6]['area_geom'].str.replace(',','.').astype('float')

In [10]:
#Vamos agora calcular o valor de soma das áreas e criar uma coluna para dividir pela total de Teles Pires
lista_colunas = ['AssentSum','CAR_Sum','FloraNDest','SigefSum','ProtIntSum','Sus_Sum','TI_Sum']
Assentamento_soma = pd_list[0][lista_colunas[0]].sum()
Car_soma = pd_list[1][lista_colunas[1]].sum()
FloraNDest =  pd_list[2][lista_colunas[2]].sum()
SigefSum =  pd_list[3][lista_colunas[3]].sum()
ProtInt = pd_list[4][lista_colunas[4]].sum()
SusSum = pd_list[5][lista_colunas[5]].sum()
TIsum = pd_list[6][lista_colunas[6]].sum()
lista_valores = [Assentamento_soma,Car_soma,FloraNDest,SigefSum,ProtInt,SusSum,TIsum]

#Pegando o numero de observaçoes para cada classe fundiária
lista_linhas = [pd_list[0].shape[0],pd_list[1].shape[0],pd_list[2].shape[0],pd_list[3].shape[0],pd_list[4].shape[0],pd_list[5].shape[0],pd_list[6].shape[0]]

#criando lista com o valor de área da geometria das malhas:
lista_geometria_area = [pd_list[0].area_geom.sum(),pd_list[1].area_geom.sum(),
                        pd_list[2].area_geom.sum(),pd_list[3].area_geom.sum(),
                        pd_list[4].area_geom.sum(),pd_list[5].area_geom.sum(),pd_list[6].area_geom.sum()]

lista_geometria_area

[35040.151534712975,
 8727.290677515366,
 354988.4136358575,
 307672.4808978155,
 266771.90703648794,
 460342.03222674434,
 364755.79044779186]

In [11]:
#criando um data frame com esses valores:
df_soma_malha = pd.DataFrame({'Tipo de Malha':lista_colunas,'Número de Propriedades':lista_linhas,'Soma, em Dolar, de importancia da Malha Fundiaria':lista_valores,
                             'Area da Malha':lista_geometria_area})
df_soma_malha['Porcentagem - Soma Por Total'] = (df_soma_malha['Soma, em Dolar, de importancia da Malha Fundiaria']/66846626)*100
df_soma_malha['Proporção - Soma Por Area da Malha'] =(df_soma_malha['Soma, em Dolar, de importancia da Malha Fundiaria']/df_soma_malha['Area da Malha'])
df_soma_malha['Proporção - Soma por Nº Propriedade'] = (df_soma_malha['Soma, em Dolar, de importancia da Malha Fundiaria'] / df_soma_malha['Número de Propriedades'])
df_soma_malha
df_soma_malha_order = df_soma_malha[['Tipo de Malha', 'Número de Propriedades', 'Soma, em Dolar, de importancia da Malha Fundiaria', 
                                     'Area da Malha','Porcentagem - Soma Por Total','Proporção - Soma Por Area da Malha','Proporção - Soma por Nº Propriedade']]
df_soma_malha_order.sort_values(by='Soma, em Dolar, de importancia da Malha Fundiaria',ascending=False)

Tipo de Malha  Número de Propriedades  \
3      SigefSum                    5330   
6        TI_Sum                      49   
2    FloraNDest                     644   
5       Sus_Sum                      70   
4    ProtIntSum                      31   
1       CAR_Sum                    1527   
0     AssentSum                     259   

   Soma, em Dolar, de importancia da Malha Fundiaria  Area da Malha  \
3                                         15740286.0  307672.480898   
6                                         15425792.0  364755.790448   
2                                         13478991.0  354988.413636   
5                                          8026222.0  460342.032227   
4                                          4550257.0  266771.907036   
1                                          2560918.0    8727.290678   
0                                           667455.0   35040.151535   

   Porcentagem - Soma Por Total  Proporção - Soma Por Area da Malha  \
3                     23.546867                           51.159226   
6                     23.076396                           42.290739   
2                     20.164056                           37.970228   
5                     12.006922                           17.435345   
4                      6.807011                           17.056732   
1                      3.831036                          293.437917   
0                      0.998487                           19.048291   

   Proporção - Soma por Nº Propriedade  
3                          2953.149343  
6                        314812.081633  
2                         20930.110248  
5                        114660.314286  
4                        146782.483871  
1                          1677.091028  
0                          2577.046332

Podemos perceber que, em dolar, os imóveis cadastrados no SIGEF somam 15.740.286 de dolares de importância para Teles Pires o que representa 23.5% do valor total para Teles Pires. <br/><br/>
Depois vem Terra Indigena com 15.425.792 de dolares de importância para Teles Pires, o que representa 23.07% para Teles Pires.<br/><br/>




### Selecionando agora os top imoveis baseado na soma e soma de área

In [84]:
#Instalando pacote:
!pip install jenkspy

     -------------------------------------- 179.2/179.2 kB 1.4 MB/s eta 0:00:00


Primeiro para área

In [12]:
import jenkspy

In [27]:
#Criando as quebras para cada malha fundiária
jenks_assentamento = jenkspy.jenks_breaks(pd_list[0]['AssentSum'], n_classes=5)

jenks_CAR = jenkspy.jenks_breaks(pd_list[1]['CAR_Sum'], n_classes=5)

jenks_FloraNaoDest = jenkspy.jenks_breaks(pd_list[2]['FloraNDest'], n_classes=5)

jenks_Sigef = jenkspy.jenks_breaks(pd_list[3]['SigefSum'], n_classes=5)

jenks_ProtInt = jenkspy.jenks_breaks(pd_list[4]['ProtIntSum'], n_classes=5)

jenks_SusSum = jenkspy.jenks_breaks(pd_list[5]['Sus_Sum'], n_classes=5)

jenks_TI = jenkspy.jenks_breaks(pd_list[6]['TI_Sum'], n_classes=5)

In [29]:
#Criando coluna para cada dataframe sobre os Buckets referentes a cada jenks pela soma dos dolares
coluna_name = ['AssentSum','CAR_Sum','FloraNDest','SigefSum','ProtIntSum','Sus_Sum','TI_Sum']
jenks=[jenks_assentamento,jenks_CAR,jenks_FloraNaoDest,jenks_Sigef,jenks_ProtInt,jenks_SusSum,jenks_TI]
for i in range(len(pd_list)):
    pd_list[i]['cut_jenks']= pd.cut(pd_list[i][coluna_name[i]],
                        bins=jenks[i],
                        labels=['bucket_1', 'bucket_2', 'bucket_3', 'bucket_4', 'bucket_5'], include_lowest=True)
 


In [33]:
#Criando lista de dataframes ja filtrado com os jenks que queremos
pd_list_jenks = []
for i in range(len(pd_list)):
    pd_list_jenks.append(pd_list[i].query('cut_jenks=="bucket_4" or cut_jenks=="bucket_5"'))



OID_                                   COD_IMOVEL             NUM_AREA  \
125    125  MA-2100501-991FFDB6C50D4FB280087F7856702EF0  152,712780111000001   
126    126  TO-1718758-2835A1127A32400CAA3DF28325CEFACF  408,273003270999993   
130    130  TO-1713205-C0CE688646394942A3376708286A07CE   41,497262403900002   
177    177  PA-1502152-AE3AB3DBCDC344158C32F49829C50156  109,268299999999996   
179    179  PA-1502152-7EEDCB7DFA8E4235A3BC685449A26636   89,153999999999996   
...    ...                                          ...                  ...   
1150  1150  MA-2104081-45AB69A5483C4E13AF689809C880F360   58,243907413499997   
1152  1152  MA-2106706-C0841D54535042899E27D728D79A743D  257,595865376000006   
1197  1197  MA-2112308-AE40C2035B584D68986D5FD7CFE6CF36    9,018682497080000   
1202  1202  MA-2109106-4E9835824C2C4ABCA1880C4E5AEC97A2   65,029779543199993   
1205  1205  MA-2104404-1E2BFE8813944AA79514F8A6C74889FE   30,476295075600000   

     COD_ESTADO             NOM_MUNICI         NUM_MODULO TIPO_IMOVE SITUACAO  \
125          MA         Alto Parnaíba  2,036200000000000        IRU       AT   
126          TO               Rio Sono  5,103400000000000        IRU       AT   
130          TO  Miracema do Tocantins  0,518700000000000        IRU       AT   
177          PA    Canaã dos Carajás  1,561000000000000        IRU       PE   
179          PA    Canaã dos Carajás  1,273600000000000        IRU       PE   
...         ...                    ...                ...        ...      ...   
1150         MA       Fernando Falcão  0,832100000000000        IRU       AT   
1152         MA                Mirador  3,679900000000000        IRU       AT   
1197         MA                 Tuntum  0,128800000000000        IRU       AT   
1202         MA       Presidente Dutra  0,929000000000000        IRU       AT   
1205         MA        Gonçalves Dias  0,435400000000000        IRU       AT   

                                             CONDICAO_I         Shape_Leng  \
125   Aguardando análise, não passível de revisã...  0,060476256396500   
126                                  Aguardando analise  0,077069521321700   
130                                  Aguardando analise  0,031604723601300   
177   ANALISADO_COM_PENDENCIAS_AGUARDANDO_RETIFICACA...  0,051428078989500   
179   ANALISADO_COM_PENDENCIAS_AGUARDANDO_RETIFICACA...  0,043895477071000   
...                                                 ...                ...   
1150  Aguardando análise, não passível de revisã...  0,029935178814300   
1152                                 Aguardando analise  0,117071572414000   
1197  Aguardando análise, não passível de revisã...  0,012668560105200   
1202                                 Aguardando analise  0,030320095947300   
1205                                Aguardando análise  0,025494749161200   

             Shape_Area  CAR_Sum  area_geom           sum_Area Malha cut_jenks  
125   0,000125783955684   4233.0   1.527952  0,000360961882308   CAR  bucket_4  
126   0,000335575639450   4723.0   4.075621  0,000862930557258   CAR  bucket_4  
130   0,000034143901109   3895.0   0.414453  0,000106406482602   CAR  bucket_4  
177   0,000089372747784   3678.0   1.093200  0,000297226717219   CAR  bucket_4  
179   0,000072923629905   4077.0   0.891989  0,000218785684688   CAR  bucket_4  
...                 ...      ...        ...                ...   ...       ...  
1150  0,000047619254790   3204.0   0.582902  0,000181929562672   CAR  bucket_4  
1152  0,000210590946684   3134.0   2.577824  0,000822534655600   CAR  bucket_4  
1197  0,000007363312842   3400.0   0.090255  0,000026545555196   CAR  bucket_4  
1202  0,000053090544587   3207.0   0.650753  0,000202916388965   CAR  bucket_4  
1205  0,000024509195322   3109.0   0.300511  0,000096658268645   CAR  bucket_4  

[151 rows x 16 columns]

In [40]:
lista_colunas = ['AssentSum','CAR_Sum','FloraNDest','SigefSum','ProtIntSum','Sus_Sum','TI_Sum']

lista_soma_ = [pd_list_jenks[0][lista_colunas[0]].sum(),pd_list_jenks[1][lista_colunas[1]].sum(),pd_list_jenks[2][lista_colunas[2]].sum(),
              pd_list_jenks[3][lista_colunas[3]].sum(),pd_list_jenks[4][lista_colunas[4]].sum(),
              pd_list_jenks[5][lista_colunas[5]].sum(),pd_list_jenks[6][lista_colunas[6]].sum()]


[148309.0, 614478.0, 8089371.0, 1681474.0, 2387703.0, 4582269.0, 7464280.0]